io/net/net_tools.py

```diff
@@ -0,0 +1,289 @@
+                self.error("%s package is need to test" % pkg)
+        self.ipv6 = False
+        if os.path.exists("/proc/net/if_inet6"):
+            self.ipv6 = True
+        interface_out = process.system_output("ip route show default",
+                                              verbose=False,
+                                              shell=True)
+        search_obj = re.search(
+            "default via\s+(\S+)\s+dev\s+(\w+)", interface_out)
+        active_iface = search_obj.group(2)
+        self.default_router = search_obj.group(1)
+        if active_iface is None:
+            self.error("can't find active network interface")
+        tmpdir = data_dir.get_tmp_dir()
+        self.backup_dir = os.path.join(tmpdir, 'net-tools')
+        if not os.path.isdir(self.backup_dir):
+            os.makedirs(self.backup_dir)
+        # Get Hostname
+        output = process.run(cmd="hostname", shell=True, verbose=False)
+        if output.exit_status != 0:
+            self.error("Failed To get hostname\n")
+        else:
+            if output.stdout is None:
+                # set hostname if not set
+                if not process.system("hostname localhost.localdomain"):
+                    self.error("Failed to set hostname\n")
+            process.system("hostname > %s/HOSTNAME" % self.backup_dir,
+                           verbose=False,
+                           shell=True)
+
+    def test_hostname(self):
+        """
+        Verify the functionality of the hostname command
+        """
+        hostname_cmd = "hostname"
+        output = process.run(cmd=hostname_cmd, shell=True, verbose=False)
+        if output.exit_status != 0 and output.stdout is None:
+            self.error("unexpected response from hostname command")
+
+        # Verifying different options provided by hostname
+        options_to_verify = self.params.get('hostname_opt', default="f")
+        for option in options_to_verify:
+            self.log.info("Verifying the hostname -%s option" % option)
+            status = process.system("hostname -%s" % option, verbose=False,
+                                    shell=True)
+            if status != 0:
+                self.error(
+                    "unexpected response from hostname %s command" %
+                    option)
+
+        # Test for setting new hostname using -F option
+        process.system(
+            "echo 'myhost.my-domain' > %s/MYHOSTNAME" %
+            self.backup_dir, verbose=False, shell=True)
+        status = process.system("hostname -F %s/MYHOSTNAME" % self.backup_dir,
+                                verbose=False,
+                                shell=True)
+        if status != 0:
+            self.error("Failed to change the hostname")
+        else:
+            output = process.run(
+                cmd=hostname_cmd, shell=True, verbose=False)
+            if not (re.search("myhost.my-domain", output.stdout)):
```

Comment:
This matches `myhost${any_char}my-domain`. I'd suggest either `if output.stdout != "myhost.my-domain"` or a bit more lenient `if "myhost.my-domain" not in output.stdout`.